src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -47,10 +46,12 @@ private MetadataGeneratorUtil() {
      * Generate metadata from the module source files available in the input argument path.
      *
      * @param args arguments
+     * @param moduleFolders List of folders to check
      * @throws IOException ioException
      * @throws CheckstyleException checkstyleException
      */
-    public static void generate(String... args) throws IOException, CheckstyleException {
+    public static void generate(List<String> moduleFolders, String... args)
```

Comment:
Since this isn't looking like a main method at all anymore, we should just change `String... args` to `String path`.

Reply:
Done